core/include/kernel/interrupt.h

```diff
@@ -164,4 +197,148 @@ static inline struct itr_handler *itr_alloc_add(size_t it,
 				       0);
 }
 
+/*
+ * Interrupt controller chip API functions
+ */
+
+/*
+ * interrupt_call_handlers() - Call registered handlers for an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ *
+ * This function is called from an interrupt context by a primary interrupt
+ * handler. This function calls the handlers registered for that interrupt.
+ * If interrupt is not handled, it is masked.
+ */
+void interrupt_call_handlers(struct itr_chip *chip, size_t itr_num);
+
+/*
+ * interrupt_mask() - Mask an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ *
+ * This function may be called in interrupt context
+ */
+static inline void interrupt_mask(struct itr_chip *chip, size_t itr_num)
+{
+	chip->ops->mask(chip, itr_num);
+}
+
+/*
+ * interrupt_unmask() - Unmask an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ *
+ * This function may be called in interrupt context
+ */
+static inline void interrupt_unmask(struct itr_chip *chip, size_t itr_num)
+{
+	chip->ops->unmask(chip, itr_num);
+}
+
+/*
+ * interrupt_enable() - Enable an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ */
+static inline void interrupt_enable(struct itr_chip *chip, size_t itr_num)
+{
+	if (chip->ops->enable)
+		chip->ops->enable(chip, itr_num);
+	else
+		interrupt_unmask(chip, itr_num);
+}
+
+/*
+ * interrupt_disable() - Disable an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ */
+static inline void interrupt_disable(struct itr_chip *chip, size_t itr_num)
+{
+	if (chip->ops->disable)
+		chip->ops->disable(chip, itr_num);
+	else
+		interrupt_mask(chip, itr_num);
+}
+
+/*
+ * interrupt_configure() - Configure an interrupt in an interrupt controller
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ * @type	Trigger type ITR_TYPE_* of ITR_TYPE_NONE
```

Comment:
s/of/or/ ?
Probably copy-paste the description from next API interrupt_configure(); it's more clear and keeps consistency. Also copy-paste for next argument "prio"